examples/compiler/iet/tut_1.ipynb

```diff
@@ -0,0 +1,1162 @@
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[2., 2., 2.],\n",
+       "       [2., 2., 2.],\n",
+       "       [2., 2., 2.]],\n",
+       "\n",
+       "      [[3., 3., 3.],\n",
+       "       [3., 3., 3.],\n",
+       "       [3., 3., 3.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "op.apply(time=2)\n",
+    "u.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is the first time that we are invoking the operator's apply. Therefore, the $kernel$ that we saw will get written in a `.c` file, and compiled into a `.so` lib if `DEVITO_BACKEND` is set to `core`.\n",
+    "\n",
+    "Then, as no key-value parameters are specified, the operator runs with its default arguments, namely `u=u, x_m=0, x_M=2, y_m=0, y_M=2,` and `time_m=0`. The subindexes `m` and `M` stands for the minimum and the maximum at those specific dimensions, respectivelly. Thus `time_M` will be set to `2`, here.\n",
+    "\n",
+    "At this point, the same operator can be used for a completely different run. Let's create another time function for governing a similar domain. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]],\n",
+       "\n",
+       "      [[0., 0., 0.],\n",
+       "       [0., 0., 0.],\n",
+       "       [0., 0., 0.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "u2 = TimeFunction(name='u', grid=grid)\n",
+    "u2.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Any of the operator default arguments may be replaced just by passing suitable key-value parameters.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Data([[[0., 0., 0.],\n",
+       "       [4., 4., 0.],\n",
+       "       [4., 4., 0.]],\n",
+       "\n",
+       "      [[0., 0., 0.],\n",
+       "       [3., 3., 0.],\n",
+       "       [3., 3., 0.]]], dtype=float32)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "op.apply(u=u2, x_m=1, x_M=2, y_m=0, y_M=1, time_M=3)\n",
+    "u2.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice, however, that there is no need for recompilation of the kernel. Just-in-time (JIT) compilation occurs only once, triggered by the first execution.\n",
```

Comment:
"Note, however, that there is no recompilation. Just-int-time (JIT) compilation ..."